utkdigitalinitiatives/utk-fedora-fedoragsearch-solr-config
# Address DIGITAL-921: Make decade facet work.

**Jira Issue**: [DIGITAL-921](https://jirautk.atlassian.net/browse/DIGITAL-921)

## What does this Pull Request do?

On October 6, I received this message came to dlcontact:

> When I click on the decade at https://digital.lib.utk.edu/collections/commencementscollection to browse, I get no search results

This is due to the template never firing.  I'm not sure why this is, but this fixes it.

## What's new?

I added a new mode called `utk_MODS_decade` and put the template in it.

## Do we really want to handle this this way?

I don't love this method, but this has been a problem for what looks like two years.  This fixes it until we get off Islandora 7.

## How should this be tested?

A description of what steps someone could take to:

1. Add an object to islandora vagrant
2. Look at its Solr document:  [http://localhost:8080/solr/collection1/select?q=PID%3A%22test:1%22&fl=*&wt=json&indent=true](http://localhost:8080/solr/collection1/select?q=PID%3A%22test:1%22&fl=*&wt=json&indent=true)
3. If overwriting a transform, replace your transform at this path: /var/lib/tomcat7/webapps/fedoragsearch/WEB-INF/classes/fgsconfigFinal/index/FgsIndex/islandora_transforms/
4. Restart tomcat or solr: http://localhost:8080/manager/
5. Update gsearch for your object with curl or python or GUI:

```python

import requests

my_pid = 'test:1'
requests.post(f'http://localhost:8080/fedorafedoragsearch/rest?operation=updateIndex&action=fromPid&value={my_pid}', auth=('fedoraAdmin', 'fedoraAdmin'))

```

You can also just make an update to MODS document on dlwork that should generate this field and check for it since I tested this there.

## Additional Notes:

Again, not ideal.  I don't like 3 modes, but there is a lot of crap going on here and we just need to correct and respond.
